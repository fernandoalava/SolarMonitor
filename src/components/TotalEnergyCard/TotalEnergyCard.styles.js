import styled from "styled-components";
import Card from "@material-ui/core/Card";

export const StyledCard = styled(Card)`
  width: 300;
  height: 150;
`;
