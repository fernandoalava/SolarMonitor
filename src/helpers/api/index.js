import config from "../../config";
import axios from "axios";

export const fetchCloudData = async (dataHandler, errorHanlder) => {
  try {
    let response = await axios.get(
      `${config.protocol}://${
        config.planetOSDomain
      }/v1/datasets/bom_access-g_global_40km/point?origin=dataset-details&apikey=${
        config.apiKey
      }&count=50&lat=49.6&lon=-50.5&var=av_ttl_cld`
    );
    dataHandler(response.data.entries);
  } catch (err) {
    errorHanlder(err);
  }
};

export const fetchSolarData = async (dataHandler, errorHanlder) => {
  try {
    let response = await axios.get(
      `${config.protocol}://${
        config.planetOSDomain
      }/v1/datasets/bom_access-g_global_40km/point?origin=dataset-details&apikey=${
        config.apiKey
      }&count=50&lat=49.6&lon=-50.5&var=av_swsfcdown`
    );
    dataHandler(response.data.entries);
  } catch (err) {
    errorHanlder(err);
  }
};
