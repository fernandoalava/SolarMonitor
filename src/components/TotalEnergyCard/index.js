import React from "react";
import PropTypes from "prop-types";

import Typography from "@material-ui/core/Typography";
import CardContent from "@material-ui/core/CardContent";
import { StyledCard } from "./TotalEnergyCard.styles";

export const TotalEnergyCard = props => {
  const totalEnergy = props.data.reduce(
    (accumulator, item) => accumulator + Number(item.wattage),
    0
  );
  return (
    <StyledCard>
      <CardContent>
        <Typography variant="h5" component="h2">
          Total Energy
        </Typography>
        <Typography variant="h3" gutterBottom>
          {(totalEnergy / 100).toFixed(2)} kW
        </Typography>
      </CardContent>
    </StyledCard>
  );
};

TotalEnergyCard.propTypes = {
  data: PropTypes.array.isRequired
};

TotalEnergyCard.defaultProps = {
  data: []
};

export default TotalEnergyCard;
