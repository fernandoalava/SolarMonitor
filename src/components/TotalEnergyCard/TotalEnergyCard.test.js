import { shallow } from "enzyme";
import * as React from "react";
import TotalEnergyCard from ".";
import { mockPanelsData } from "../../helpers";

describe("TotalEnergyCard", () => {
  it("Renders correctly without data", () => {
    const wrapper = shallow(<TotalEnergyCard data={[]} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("Renders correctly with data", () => {
    const wrapper = shallow(<TotalEnergyCard data={mockPanelsData} />);
    expect(wrapper).toMatchSnapshot();
  });
});
