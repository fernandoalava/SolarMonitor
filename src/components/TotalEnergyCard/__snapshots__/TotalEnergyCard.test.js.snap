// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TotalEnergyCard Renders correctly with data 1`] = `
<Styled(WithStyles(ForwardRef(Card)))>
  <WithStyles(ForwardRef(CardContent))>
    <WithStyles(ForwardRef(Typography))
      component="h2"
      variant="h5"
    >
      Total Energy
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))
      gutterBottom={true}
      variant="h3"
    >
      119.77
       kW
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(CardContent))>
</Styled(WithStyles(ForwardRef(Card)))>
`;

exports[`TotalEnergyCard Renders correctly without data 1`] = `
<Styled(WithStyles(ForwardRef(Card)))>
  <WithStyles(ForwardRef(CardContent))>
    <WithStyles(ForwardRef(Typography))
      component="h2"
      variant="h5"
    >
      Total Energy
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))
      gutterBottom={true}
      variant="h3"
    >
      0.00
       kW
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(CardContent))>
</Styled(WithStyles(ForwardRef(Card)))>
`;
